indolaw-parser/parser_utils.py

```diff
@@ -880,21 +972,29 @@ def clean_split_plaintext(law: List[str]) -> List[str]:
     print_section_header('CLEANING SPLIT PLAINTEXT...')
 
     new_law: List[str] = []
-    for i, line in enumerate(law):
+
+    idx = 0
+    last_idx = len(law)
+    # idx_input contains dicts of input index and the associated input
+    idx_input = []
+    
+    while idx < last_idx:
         '''
         the line length check is a heuristic to filter out false positives from the
         lowercase check due to list indexes e.g 'e.'
 
         The logic below is imprecise; it's just all heuristics & hands off to the user
         to make a decision
         '''
-        really_long = len(law[i]) > 75
-        long_enough = len(law[i]) > 5
-        starts_with_lowercase = law[i][0].islower()
-        starts_with_number = law[i][0].isnumeric()
-        previous_line_long = i > 0 and len(law[i-1]) > 20
-        previous_line_not_all_caps = i > 0 and not law[i-1].isupper()
-        not_all_caps = not law[i].isupper()
+        line = law[idx]
+
+        really_long = len(law[idx]) > 75
+        long_enough = len(law[idx]) > 5
+        starts_with_lowercase = law[idx][0].islower()
+        starts_with_number = law[idx][0].isnumeric()
+        previous_line_long = idx > 0 and len(law[idx-1]) > 20
+        previous_line_not_all_caps = idx > 0 and not law[idx-1].isupper()
+        not_all_caps = not law[idx].isupper()
```

Comment:
since you already have `line` can you replace all uses of `law[idx]` below with `line`?